src/base/bittorrent/torrenthandle.h

```diff
@@ -445,14 +445,26 @@ namespace BitTorrent
 
         QHash<QString, TrackerInfo> m_trackerInfos;
 
-        enum StartupState
+        enum ProcessingState
         {
-            NotStarted,
-            Starting,
-            Started
+            PreparingPaused, // torrent is preparing to start processing in "paused" state
+            PreparingResumed, // torrent is preparing to start processing in "resumed" state
```

Comment:
The comment is confusing to me, can a torrent have resumed state when it is still in preparing stage?

Reply:
I'm sorry, but it's really hard to point out the differences between high/low level states, because they're called the same...

TorrentHandle instance can be created either in Paused or Resumed user-level state (e.g. it was Paused when user close qBittorrent last time so it should be Paused after preparations). But torrent can be actually in different native state during its preparation. We need to start it resumed to allow libtorrent checking it, then it should be paused to perform other preparations. Then Resumed torrent should be actually resumed. And we need to handle all these native state transitions properly behind the scenes.